tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TypeSpec/TspClientUpdateTool.cs

```diff
@@ -0,0 +1,151 @@
+
+    public TspClientUpdateTool(ILogger<TspClientUpdateTool> logger, IOutputHelper output, Func<string, IUpdateLanguageService> languageServiceFactory)
+    {
+        this.logger = logger;
+        this.output = output;
+        this.languageServiceFactory = languageServiceFactory;
+        CommandHierarchy = [ SharedCommandGroups.TypeSpec ];
+    }
+
+    public override Command GetCommand()
+    {
+        var cmd = new Command("customized-update",
+            description: "Update customized TypeSpec-generated client code. Runs the full pipeline by default: regenerate -> diff -> map -> propose -> apply");
+        cmd.AddArgument(specPathArg);
+        cmd.AddOption(SharedOptions.PackagePath);
+        cmd.AddOption(newGenOpt);
+        cmd.SetHandler(async ctx => await HandleUnified(ctx, ctx.GetCancellationToken()));
+        return cmd;
+    }
+
+    public override Task HandleCommand(InvocationContext ctx, CancellationToken ct) => Task.CompletedTask;
+
+    [McpServerTool(Name = "azsdk_tsp_update"), Description("Update pipeline")] 
+    public async Task<TspClientUpdateResponse> UnifiedUpdateAsync(string specPath, IUpdateLanguageService languageService, CancellationToken ct = default)
+    {
+        try
+        {
+            if (string.IsNullOrWhiteSpace(specPath))
+            {
+                return new TspClientUpdateResponse { ResponseError = "Spec path required", ErrorCode = "MissingSpec" };
+            }
+            var session = new UpdateSessionState { SpecPath = specPath };
+
+            // Regenerate (placeholder)
+            session.LastStage = UpdateStage.Regenerated;
+            session.Status = "Regenerated";
+
+            // Diff
+            var apiChanges = await languageService.DiffAsync(new Dictionary<string, SymbolInfo>(), new Dictionary<string, SymbolInfo>());
+            session.ApiChanges = apiChanges;
+            session.ApiChangeCount = apiChanges.Count;
+            session.LastStage = UpdateStage.Diffed;
+            session.Status = "Diffed";
+            // Continue pipeline even if no API changes so validation & auto-fix logic can run.
+
+            // Map
+            var customizationRoot = session.NewGeneratedPath; // placeholder path usage
+            var impacts = await languageService.AnalyzeCustomizationImpactAsync(session, customizationRoot, session.ApiChanges, ct);
+            session.ImpactedCustomizations = impacts;
+            session.LastStage = UpdateStage.Mapped;
+            session.Status = "Mapped";
+            // Even if no impacts, continue through remaining stages for a consistent single-pass experience.
+            if (impacts.Count == 0)
+            {
+                session.Status = "NoCustomizationImpact"; // informational; continue
+            }
+
+            // Propose
+            var patches = await languageService.ProposePatchesAsync(session, impacts, ct);
```

Comment:
Is this required if there is no impact? It will simply run and make no changes. right?

Reply:
It is a no-op basically in that case, teh unified still runs e2e.